src/box-node-sdk.ts

```diff
@@ -0,0 +1,463 @@
+	Webhooks = require('./managers/webhooks');
+
+// ------------------------------------------------------------------------------
+// Private
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+/**
+ * A backend NodeJS SDK to interact with the Box V2 API.
+ * This is the single entry point for all SDK consumer interactions. This is the only file that a 3rd party app
+ * should require. All other components are private and reached out to via this component.
+ * 1. Provides getters to spawn client instances for users to interact with the Box API.
+ * 2. Provides manual capability to acquire tokens via token grant endpoints.
+ *    However, it is recommended to use clients to do this for you.
+ * 3. Emits notification events about relevant request/response events. Useful for logging Box API interactions.
+ *    Notification events: request retries, exceeding max retries, permanent failures.
+ *
+ * @param {UserConfigurationOptions} params User settings used to initialize and customize the SDK
+ * @constructor
+ */
+class BoxSDKNode extends EventEmitter {
+	accessLevels: any /* FIXME */;
+	collaborationRoles: any /* FIXME */;
+	CURRENT_USER_ID: any /* FIXME */;
+	config: any /* FIXME */;
+	_eventBus!: EventEmitter;
+
+	requestManager!: APIRequestManager;
+	tokenManager!: TokenManager;
+	anonymousSession!: AnonymousAPISession;
+
+	/**
+	 * Expose the BoxClient property enumerations to the SDK as a whole. This allows
+	 * the consumer to access and use these values from anywhere in their application
+	 * (like a helper) by requiring the SDK, instead of needing to pass the client.
+	 */
+	static accessLevels = BoxSDKNode.prototype.accessLevels;
+	static collaborationRoles = BoxSDKNode.prototype.collaborationRoles;
+	static CURRENT_USER_ID = BoxSDKNode.prototype.CURRENT_USER_ID;
+
+	/**
+	 * Expose Webhooks.validateMessage() to the SDK as a whole. This allows
+	 * the consumer to call BoxSDK.validateWebhookMessage() by just requiring the SDK,
+	 * instead of needing to create a client (which is not needed to validate messages).
+	 */
+	static validateWebhookMessage = Webhooks.validateMessage;
+
+	constructor(params: UserConfigurationOptions) {
+		super();
+
+		var eventBus = new EventEmitter();
+
+		var self = this;
+		eventBus.on('response', function () {
+			var args: any /* FIXME */ = [].slice.call(arguments);
+			args.unshift('response');
+			self.emit.apply(self, args);
+		});
+
+		// Setup the configuration with the given params
+		this.config = new Config(params);
+		this._eventBus = eventBus;
+		this._setup();
+	}
+
+	/**
+	 * Setup the SDK instance by instantiating necessary objects with current
+	 * configuration values.
+	 *
+	 * @param {EventEmitter} eventBus The event bus to use
```

Comment:
I see that this is carried over from the original file, so if this is a bug it's probably an old bug 🤔   But it looks like `_setup` doesn't take any params. Should we remove this line?